### PR TITLE
Simplify and modernise templates

### PR DESCRIPTION
**Pull request by galderz**
~Don't integrate yet.~

Persistent template not yet working ([permission error](https://gist.github.com/galderz/9ac35d976ed5841bffdbdb50a85e03d6)).

Moved templates to yaml where the XML in the ConfigMap can be seen as is, rather than the convoluted escaped text in previous json templates. 

Add very basic tests to see if templates work.

Improved readiness of ephemeral template, which was taking several minutes to be ready when in fact the server is ready after 10-15 seconds.

README still needs updating. I'll leave that until the end.

### CODE REVIEW

**Comment by galderz** [comment]
@gustavonalle @slaskawi Persistent templates fails with [this error](https://gist.github.com/galderz/9ac35d976ed5841bffdbdb50a85e03d6). It appears to be a permission error. I tried with 9.4.1 image and still has the same issue. Our dockerfile already does some adjustments ([here](https://github.com/jboss-dockerfiles/infinispan/blob/master/server/Dockerfile#L34-L35)), anything that we should be doing differently? The template section that handles server data folder is identical to downstream.

**Comment by slaskawi** [comment]
@galderz Unfortunately I've never seen this error. Maybe it just can't create a subdirectory (`data/content`). You can use `oc rsh` to check that...

File: test/test-persistent.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    ready=$(isPodReady)
+    status=$(getPodStatus)
+    echo "Pod: ready=${ready},status=${status}"
+    sleep 10
+  done
+}
+
+waitForReady
+
+#getRouteHost() {
```
Comment (ryanemerson) [inline test/test-persistent.sh]: Why is this commented?

Comment (galderz) [reply test/test-persistent.sh]: Because the persistent test does not yet work.

**Pull request merged**